backend/molgenis-emx2-webapi/src/main/java/org/molgenis/emx2/web/FileApi.java

```diff
@@ -54,17 +54,22 @@ public static String getFile(Request request, Response response) throws IOExcept
       throw new MolgenisException(
           "Download failed: file id '" + id + "' not found in table " + tableName);
     }
-    String ext = result.get(0).getString(columnName + "_extension");
-    String mimetype = result.get(0).getString(columnName + "_mimetype");
-    byte[] contents = result.get(0).getBinary(columnName + "_contents");
+    addFileColumnToResponse(response, columnName, result.get(0));
+    return "";
+  }
+
+  public static void addFileColumnToResponse(Response response, String columnName, Row row)
+      throws IOException {
+    String ext = row.getString(columnName + "_extension");
+    String mimetype = row.getString(columnName + "_mimetype");
```

Comment:
is there a check on insert which mimetypes and extensions are valid? And is there a check for the name, so it does not do path-traversal-exploits etc?

Reply:
the name cannot be defined by the user but instead is given by emx2. The only thing user can provide is the extension, which is appended so I believe cannot be abused?

Reply:
that will work!